reconcile/openshift_base.py

```diff
@@ -511,6 +515,9 @@ def validate_data(oc_map, actions):
                     if conditions:
                         logging.info(f'Job conditions are: {conditions}')
                         logging.info(yaml.safe_dump(conditions))
+                        for c in conditions:
+                            if c.get('type') == 'Failed':
+                                raise ValidationErrorJobFailed(name)
```

Comment:
I tend to prefer this style:

```
if any(c.get('type') == 'Failed') for c in conditions):
    raise...
```
Also, is there any other information about this failed job that may help the on call/IC troubleshoot problems? I really like descriptive exceptions.

Reply:
added the reason field. fwiw, this will mostly not be for IC/oncall to debug, but a result of a deployment of a service, which will be debugged by the tenants. but a reason makes sense nonetheless.